docs/tutorials/tutorial_start_EOS-S3_HW.md

```diff
@@ -0,0 +1,98 @@
+(parts *Docker image*, *Building*).
+
+> Note: There was problem during compilation for Cortex-M4 device. 
+
+```shell
+make -f makefiles/Makefile_Startup
+make[2]: Entering directory '/home/qorc-sdk/workspace/qorc-onion-apps/qorc_fpga_breathectrl/GCC_Project'
+make[2]: *** No rule to make target '/home/qorc-sdk/workspace/qorc-onion-apps/qorc_fpga_breathectrl/GCC_Project/output/startup_EOSS3B_GCC.o', needed by 'all'.  Stop.
+make[2]: Leaving directory '/home/qorc-sdk/workspace/qorc-onion-apps/qorc_fpga_breathectrl/GCC_Project'
+make[1]: *** [Makefile:59: Startup] Error 2
+make[1]: Leaving directory '/home/qorc-sdk/workspace/qorc-onion-apps/qorc_fpga_breathectrl/GCC_Project'
+make: *** [Makefile:54: m4] Error 2
+```
+
+There is workaround for this problem. During start of the Docker container
+one option should be corrected:
+```
+ -v $PWD:/home/qorc-sdk/workspace
+```
+to 
+```
+ -v $PWD:/home/qorc-sdk/qorc-sdk/workspace 
+```
+
+Warning: this correction reorganizes structure of the catalog in Docker container.
+
+More generic solution is to pay attention about nesting the directories. 
+
+## Architecture overview
+
+Architectural overview of this SoC has been depicted in document: 
+[EOS S3 Datasheet](https://www.quicklogic.com/wp-content/uploads/2020/12/QL-EOS-S3-Ultra-Low-Power-multicore-MCU-Datasheet-2020.pdf), page 19.
+ 
+There is also different block diagram of the S3, with main subsystems,
+printed in [QL S3 Technical Reference Manual](https://www.quicklogic.com/wp-content/uploads/2020/06/QL-S3-Technical-Reference-Manual.pdf), page 23. 
+
+## Example code
+
+There is example applications, in :[Source](https://github.com/coolbreeze413/qorc-onion-apps/tree/master/). 
+Bunch of them, with pattern: `qorc_helloworld*` are the examples of basic
+snippets, which we could use as a template for further applications.
+
+Currently, as example, I use `qorc_helloworldm4` app. This is
+FreeRTOS-compatible, fully working snippet. 
+
+This program uses the following modules:
+* Basic ARM Cortex M4 initialization;
+* Physical interfaces:
+  * UART
+  * I2C
+* GPIO, RGB LED diode
+
+### Pin mapping
+
+Entry point for this application is [main.c](https://github.com/coolbreeze413/qorc-onion-apps/blob/master/qorc_helloworldm4/src/main.c).
+Assigning the signal to pad has been done in the [qorc_hardwareSetup() function](https://github.com/coolbreeze413/qorc-onion-apps/blob/a960ca3f450fafde9da07547b35efe8e1caa2574/qorc_helloworldm4/src/qorc_hardwaresetup.c#LL48C9-L48C9).
+To proper asignment the array `PadConfig pincfg_table[]`, passed as first
+argument should be initialized [code here](https://github.com/coolbreeze413/qorc-onion-apps/blob/a960ca3f450fafde9da07547b35efe8e1caa2574/qorc_helloworldm4/src/pincfg_table.c#L21)
+
+Assigning pin to FPGA structure could be performed as in the following [example](https://github.com/coolbreeze413/qorc-onion-apps/blob/a960ca3f450fafde9da07547b35efe8e1caa2574/qorc_helloworldfpga/fpga/rtl/quickfeather.pcf#LL44C8-L44C8)
```

Comment:
Is this enough? M4 part of SoC must be told to not drive those lines somehow, or not told to drive if that is the default state. This must be clearly stated here.